core/src/main/java/org/elasticsearch/index/analysis/AnalysisRegistry.java

```diff
@@ -271,6 +271,9 @@ public boolean requiresAnalysisSettings() {
             Settings currentSettings = entry.getValue();
             String typeName = currentSettings.get("type");
             if (analyzer) {
+                if(defaultInstance.containsKey(name)) {
+                    throw new IllegalArgumentException("do not allow custom analyzers to have the same names [" + name + "] as built-in analyzers");
```

Comment:
Maybe rephrase to something like this: "Custom analysis component [" + name + "] may not reuse the name of a built-in component"